streams/src/test/java/org/apache/kafka/streams/processor/internals/StateConsumerTest.java

```diff
@@ -0,0 +1,165 @@
+
+
+public class StateConsumerTest {
+
+    private static final long FLUSH_INTERVAL = 1000L;
+    private final TopicPartition topicOne = new TopicPartition("topic-one", 1);
+    private final TopicPartition topicTwo = new TopicPartition("topic-two", 1);
+    private final MockTime time = new MockTime();
+    private final MockConsumer<byte[], byte[]> consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+    private final Map<TopicPartition, Long> partitionOffsets = new HashMap<>();
+    private GlobalStreamThread.StateConsumer stateConsumer;
+    private StateMaintainerStub stateMaintainer;
+
+    @Before
+    public void setUp() throws Exception {
+        partitionOffsets.put(topicOne, 20L);
+        partitionOffsets.put(topicTwo, 30L);
+        stateMaintainer = new StateMaintainerStub(partitionOffsets);
+        stateConsumer = new GlobalStreamThread.StateConsumer(consumer, stateMaintainer, time, 10L, FLUSH_INTERVAL);
+    }
+
+    @Test
+    public void shouldAssignPartitionsToConsumer() throws Exception {
+        stateConsumer.initialize();
+        assertEquals(Utils.mkSet(topicOne, topicTwo), consumer.assignment());
+    }
+
+    @Test
+    public void shouldSeekToInitialOffsets() throws Exception {
+        stateConsumer.initialize();
+        assertEquals(20L, consumer.position(topicOne));
+        assertEquals(30L, consumer.position(topicTwo));
+    }
+
+    @Test
+    public void shouldUpdateStateWithReceivedRecordsForPartition() throws Exception {
+        stateConsumer.initialize();
+        consumer.addRecord(new ConsumerRecord<>("topic-one", 1, 20L, new byte[0], new byte[0]));
+        consumer.addRecord(new ConsumerRecord<>("topic-one", 1, 21L, new byte[0], new byte[0]));
+        stateConsumer.pollAndUpdate();
+        assertEquals(2, stateMaintainer.updatedPartitions.get(topicOne).intValue());
+    }
+
+    @Test
+    public void shouldUpdateStateWithReceivedRecordsForAllTopicPartition() throws Exception {
+        stateConsumer.initialize();
+        consumer.addRecord(new ConsumerRecord<>("topic-one", 1, 20L, new byte[0], new byte[0]));
+        consumer.addRecord(new ConsumerRecord<>("topic-two", 1, 31L, new byte[0], new byte[0]));
+        consumer.addRecord(new ConsumerRecord<>("topic-two", 1, 32L, new byte[0], new byte[0]));
+        stateConsumer.pollAndUpdate();
+        assertEquals(1, stateMaintainer.updatedPartitions.get(topicOne).intValue());
+        assertEquals(2, stateMaintainer.updatedPartitions.get(topicTwo).intValue());
+    }
+
+    @Test
+    public void shouldFlushStoreWhenFlushIntervalHasLapsed() throws Exception {
+        stateConsumer.initialize();
+        consumer.addRecord(new ConsumerRecord<>("topic-one", 1, 20L, new byte[0], new byte[0]));
+        time.sleep(FLUSH_INTERVAL);
+
+        stateConsumer.pollAndUpdate();
+        assertTrue(stateMaintainer.flushed);
+    }
+
+    @Test
+    public void shouldNotFlushOffsetsWhenFlushIntervalHasNotLapsed() throws Exception {
```

Comment:
Unify `shouldFlushStoreWhenFlushIntervalHasLapsed` and `shouldNotFlushOffsetsWhenFlushIntervalHasNotLapsed` ?